packages/core/src/browser/view-container.ts

```diff
@@ -76,6 +76,20 @@ export namespace BadgeWidget {
     }
 }
 
+/**
+ * A widget that may change it's internal structure dynamically. Current use is for
+ * updating the toolbar when a contributed view is contructed "lazily"
+ */
+export interface DynamicWidget {
+    onChanged: CommonEvent<void>;
```

Comment:
I really like the idea of having a dynamic widget that is able to trigger an update on it's own toolbar but I was wondering about some things:
- Isn't it conceptually very close to the `onDidChange` event on the `ConditionalToolbarItem`, is there any way to re-use that or is it because of the indirection through the ViewContainerPart?
- The name is quite generic even through it has a very specific use case (at the moment), do we really want to rely on such a generic name that might be inadvertently already implemented on a widget? Also I believe we should follow the `onDidX` naming scheme here, according to the [Coding Guidelines](https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines).

Reply:
The answer to both, IMO, is no. What's changing here is not a single toolbar item, but the set of toolbar items that apply for this view. The concrete case I'm covering is that the "real" widget that forms the context for action enablement is created lazily and we need to update the toolbar when that happens. However, that lazy creation is happening inside the view widget, and is an implementation detail of the view widget which I don't want to expose. 
I know the terminology is vague, but intentionally so. Once have more cases and understand the issue better, a better naming may become clear.